weaviate/replication/base.py

```diff
@@ -0,0 +1,66 @@
+import uuid
+from typing import Generic, Optional
+
+from httpx import Response
+
+from weaviate.connect import executor
+from weaviate.connect.v4 import ConnectionType, _ExpectedStatusCodes
+from weaviate.replication.models import (
+    ShardingState,
+    TransferType,
+)
+
+
+class _ReplicationExecutor(Generic[ConnectionType]):
+    def __init__(self, connection: ConnectionType):
+        self._connection = connection
+
+    def replicate(
+        self,
+        *,
+        collection: str,
+        shard: str,
+        source_node: str,
+        target_node: str,
+        transfer_type: TransferType = TransferType.COPY,
+    ) -> executor.Result[uuid.UUID]:
+        def resp(response: Response) -> uuid.UUID:
+            return uuid.UUID(response.json()["id"])
+
+        body = {
+            "collectionId": collection,
+            "shardId": shard,
+            "sourceNodeName": source_node,
+            "destinationNodeName": target_node,
+            "transferType": transfer_type.value,
+        }
+        return executor.execute(
+            response_callback=resp,
+            method=self._connection.post,
+            path="/replication/replicate",
+            weaviate_object=body,
+            status_codes=_ExpectedStatusCodes(200, "replicate replicate"),
+            error_msg="Failed to replicate shard",
+        )
+
+    def query_sharding_state(
+        self,
+        *,
+        collection: str,
+        shard: Optional[str],
```

Comment:
should this have the default None?

Reply:
Ah, because of the specific pattern I'm adding in `weaviate.replication`, it doesn't use the stubs-auto-gen logic instead spelling out the `async_.py` and `sync.py` files explicitly, the defaults go in those files with the executor not permitting them. I will remove the default from `transfer_type` above to make it clearer